### ISSUE TITLE
SA1118 can't be avoided in the case of disabling "this" in extension methods in .NET older than 3.5

### ISSUE THREAD

**Issue by peteroupc** [StyleCop/StyleCop#226]
The SA1118 warning is issued for parameters that span more than one line.  However, I see no way to avoid or disable this warning in the case of enabling the "this" keyword, which signals extension methods, only in .NET Framework 3.5 and later iterations of .NET.  Indeed, parts of the warning text are out of place in parameter _declarations_, especially the suggestion to "save the contents of the parameter as a temporary variable and pass the temporary variable as a parameter".

Example:

```c#
    public static int MyExtensionMethod(
    #if !NET20
       this
    #endif
       IMyInterface iface,
       int param1,
       int param2){
       // code goes here
    }
```

**Comment by peteroupc** [comment]
Withdrawn.  The new StyleCop Analyzers contains a revision of this rule that allows the first parameter, including the "this" in extension methods here, to span multiple lines.
